spring-cloud-sleuth-core/src/test/java/org/springframework/cloud/sleuth/instrument/quartz/TracingJobListenerTest.java

```diff
@@ -0,0 +1,299 @@
+
+/**
+ * @author Branden Cash
+ */
+public class TracingJobListenerTest {
+
+	private static final JobKey SUCCESSFUL_JOB_KEY = new JobKey("SuccessfulJob");
+
+	private static final JobKey EXCEPTIONAL_JOB_KEY = new JobKey("ExceptionalJob");
+
+	private static final TriggerKey TRIGGER_KEY = new TriggerKey("ExampleTrigger");
+
+	private TracingJobListener listener;
+
+	private Tracing tracing;
+
+	private Scheduler scheduler;
+
+	private CompletableFuture completableJob;
+
+	private Queue<Span> spans = new ArrayDeque<>();
+
+	@Before
+	public void setUp() throws Exception {
+		tracing = Tracing.newBuilder().spanReporter(spans::add)
+				.currentTraceContext(ThreadLocalCurrentTraceContext.newBuilder()
+						.addScopeDecorator(StrictScopeDecorator.create()).build())
+				.build();
+
+		listener = new TracingJobListener(tracing);
+		completableJob = new CompleteableTriggerListener();
+
+		scheduler = createScheduler(getClass().getSimpleName(), 1);
+		scheduler.addJob(newJob(ExceptionalJob.class).withIdentity(EXCEPTIONAL_JOB_KEY)
+				.storeDurably().build(), true);
+		scheduler.addJob(newJob(SuccessfulJob.class).withIdentity(SUCCESSFUL_JOB_KEY)
+				.storeDurably().build(), true);
+
+		scheduler.getListenerManager().addTriggerListener(listener);
+		scheduler.getListenerManager().addJobListener(listener);
+		scheduler.getListenerManager()
+				.addTriggerListener((CompleteableTriggerListener) completableJob);
+		scheduler.getListenerManager()
+				.addJobListener((CompleteableTriggerListener) completableJob);
+		scheduler.start();
+	}
+
+	@After
+	public void tearDown() throws Exception {
+		scheduler.shutdown(true);
+	}
+
+	@Test
+	public void getName_When_AllCases_Expect_ToReturnClassName() {
+		assertThat(listener.getName()).isEqualTo(TracingJobListener.class.getName());
+	}
+
+	@Test
+	public void scheduleJob_When_JobIsSuccessful_Expect_SpanToComplete()
+			throws Exception {
+		scheduler.scheduleJob(newTrigger().forJob(SUCCESSFUL_JOB_KEY).startNow().build());
+		completableJob.join();
+		takeSpan();
+	}
+
+	@Test
+	public void scheduleJob_When_JobIsSuccessful_Expect_HaveSpanWithProperNameAndTag()
+			throws Exception {
+		scheduler.scheduleJob(newTrigger().withIdentity(TRIGGER_KEY)
+				.forJob(SUCCESSFUL_JOB_KEY).startNow().build());
+		completableJob.join();
+		Span span = takeSpan();
+		assertThat(span.name()).isEqualToIgnoringCase(SUCCESSFUL_JOB_KEY.toString());
+		assertThat(span.tags().get(TRIGGER_TAG_KEY))
+				.isEqualToIgnoringCase(TRIGGER_KEY.toString());
+	}
+
+	@Test
+	public void scheduleJob_When_JobIsExceptional_Expect_SpanToComplete()
+			throws Exception {
+		scheduler
+				.scheduleJob(newTrigger().forJob(EXCEPTIONAL_JOB_KEY).startNow().build());
+		completableJob.join();
+		takeSpan();
+	}
+
+	@Test
+	public void scheduleJob_When_JobIsVetoed_Expect_SpanToComplete() throws Exception {
+		scheduler.getListenerManager().addTriggerListener(new VetoJobTriggerListener());
```

Comment:
Throughout all of these tests can we have the structure as follows?

```
// given
input conditions

// when 
a single line where sth triggers the behavior we want to assert

// then
assert the outcome
```

You can even leave the `//given, // when, // then` comments in the test code